diff --git a/lib/chart/bar/BarView.js b/lib/chart/bar/BarView.js
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/lib/chart/bar/BarView.js
+++ b/lib/chart/bar/BarView.js
@@ -130,6 +130,9 @@
     var oldData = this._data;
     var coord = seriesModel.coordinateSystem;
+    if (!coord) {
+      return;
+    }
     var baseAxis = coord.getBaseAxis();
     var isHorizontalOrRadial;
diff --git a/lib/chart/line/LineView.js b/lib/chart/line/LineView.js
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/lib/chart/line/LineView.js
+++ b/lib/chart/line/LineView.js
@@ -492,6 +492,9 @@
   LineView.prototype.render = function (seriesModel, ecModel, api) {
     var coordSys = seriesModel.coordinateSystem;
+    if (!coordSys) {
+      return;
+    }
     var group = this.group;
@@ -774,7 +777,7 @@
   LineView.prototype._changePolyState = function (toState) {
     var polygon = this._polygon;
-    setStatesFlag(this._polyline, toState);
+    this._polyline && setStatesFlag(this._polyline, toState);
     polygon && setStatesFlag(polygon, toState);
   };
diff --git a/lib/layout/barPolar.js b/lib/layout/barPolar.js
index e69f0ae7c7666d22997447bacac38d2815e48187..0813580df639c964e0bcde13aa202fa63ba769e1 100644
--- a/lib/layout/barPolar.js
+++ b/lib/layout/barPolar.js
@@ -57,7 +57,7 @@ function barLayoutPolar(seriesType, ecModel, api) {
   }));
   ecModel.eachSeriesByType(seriesType, function (seriesModel) {
     // Check series coordinate, do layout for polar only
-    if (seriesModel.coordinateSystem.type !== 'polar') {
+    if (!seriesModel.coordinateSystem || seriesModel.coordinateSystem.type !== 'polar') {
       return;
     }
     var data = seriesModel.getData();
